### PR TITLE
Python solution for week2

### PR DESCRIPTION
**Pull request by hienhpss**

### CODE REVIEW

File: week0002/hienhpss/python/week0002.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+def spi_round(n=4,m=4,x=0):
+	'''Looping through the spiral path to return the next
+	cell'''
+	if (x>=(n+1)/2) or (x>=(m+1)/2):
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: Spacing please! Give some breathing space for operators

**Comment by vietlq** [comment]
Please remove the *.pyc file too. It's not part of submission.

File: week0002/hienhpss/python/week0002.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		return
+	for i in range(x,m-x-1):
+		yield (x,i)
+	for i in range(x,n-x-1):
+		yield(i,m-x-1)
+	for i in range(m-x-1,x,-1):
+		yield(n-x-1,i)
+	for i in range(n-x-1,x,-1):
+		yield(i,x)
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: Avoid generators for now :) Why complicating the issue when you can just use loops? 🍡

File: week0002/hienhpss/python/week0002.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+def spiral(n=4,m=4):
+	'''Return the n*m board with numbers
+	arrange in spiral format'''
+	board = [[False for x in range(m)] for y in range(n)]
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: Also: `m*[n*[False]]`

**Comment by hienhpss** [comment]
Fixed the spacing issue and removed pyc file. The generator thing is a personal choice towards writing more pythonic code (and it works better for bigger problem compare with initializing a huge list).

File: week0002/hienhpss/python/week0002.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+def spi_round(n=4,m=4,x=0):
+	'''Looping through the spiral path to return the next
+	cell'''
+	if (x >= (n+1)/2) or (x >= (m+1)/2):
+		return
+	for i in range(x,m-x-1):
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: Spacing please. All operators: `+ - * / % == , =` should have space around them, except `=` in function argument passing.

File: week0002/hienhpss/python/week0002.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	'''Return the n*m board with numbers
+	arrange in spiral format'''
+	board = [[False for x in range(m)] for y in range(n)]
+	limit = int((n+1) / 2)
+	count = 1
+	for i in range(limit):
+		for next_row,next_col in spi_round(n,m,i):
+			if board[next_row][next_col] == False:
+				board[next_row][next_col] = count
+				count+=1
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: Space around `+= -= /= *=` too

File: week0002/hienhpss/python/week0002.py (hunk 4 above)
Comment (hienhpss) [inline week0002/hienhpss/python/week0002.py]: Done

File: week0002/hienhpss/python/week0002.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+def spi_round(n=4,m=4,x=0):
```
Comment (hienhpss) [inline week0002/hienhpss/python/week0002.py]: Done

File: week0002/hienhpss/python/week0002.py (hunk 5 above)
Comment (hienhpss) [inline week0002/hienhpss/python/week0002.py]: Done

File: week0002/hienhpss/python/week0002.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+def spiral(n=4, m=4):
+	'''Return the n*m board with numbers
+	arrange in spiral format'''
+	board = [[False for x in range(m)] for y in range(n)]
+	limit = int((n+1) / 2)
+	count = 1
+	for i in range(limit):
+		for next_row, next_col in spi_round(n, m, i):
+			if board[next_row][next_col] == False:
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: More Pythonic way to do it: `if not board[next_row][next_col]:`

File: week0002/hienhpss/python/week0002.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
+def spi_round(n=4, m=4, x=0):
+	'''Looping through the spiral path to return the next
+	cell'''
+	if (x >= (n+1) / 2) or (x >= (m+1) / 2):
+		return
+	for i in range(x, m-x-1):
+		yield (x, i)
+	for i in range(x, n-x-1):
+		yield(i, m-x-1)
```
Comment (vietlq) [inline week0002/hienhpss/python/week0002.py]: Spacing around `-` too.

Comment (hienhpss) [inline week0002/hienhpss/python/week0002.py]: Done

**Pull request merged**